src/main/kotlin/it/pagopa/wallet/services/WalletService.kt

```diff
@@ -41,4 +46,45 @@ class WalletService(@Autowired private val walletRepository: WalletRepository) {
             LoggedAction(wallet, WalletAddedEvent(it.id))
         }
     }
+
+    fun patchWallet(
+        walletId: UUID,
+        service: Pair<ServiceName, ServiceStatus>
+    ): Mono<LoggedAction<Wallet>> {
+        return walletRepository
+            .findById(walletId.toString())
+            .switchIfEmpty { Mono.error(WalletNotFoundException(WalletId(walletId))) }
+            .map { updateServiceList(it, service) }
+            .flatMap { walletRepository.save(it) }
+            .map { LoggedAction(it.toDomain(), WalletPatchEvent(it.id)) }
+    }
+
+    private fun updateServiceList(
+        wallet: it.pagopa.wallet.documents.wallets.Wallet,
+        service: Pair<ServiceName, ServiceStatus>
+    ): it.pagopa.wallet.documents.wallets.Wallet {
+        val updatedServiceList = wallet.services.toMutableList()
+        when (val index = wallet.services.indexOfFirst { s -> s.name == service.first.name }) {
```

Comment:
Are we sure that we want to add a service when patching if the service is not present?

Reply:
Good point. I'm not really sure about this, but, think about the use case when a new service is added to the service repo and a single user want to add it to his wallet. In this case we need this logic. Anyway we could think about having different logic for this 2 usecase, while They are logically similar from the point of view of "patch the wallet" action.